### PR TITLE
Remove ActiveSupport dependency

### PR DESCRIPTION
**Pull request by matthewshafer**
There were a few places around circuitbox where some active support methods were used, but there was not that many left. I switched those to use methods that come part of ruby. Also added a null notifier that is the default if active support notifications are not defined.

### CODE REVIEW

File: lib/circuitbox/circuit_breaker.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def initialize(service, options = {})
       @notifier = options.fetch(:notifier) { Circuitbox.default_notifier }
 
       @exceptions = options.fetch(:exceptions) { [] }
-      @exceptions = [Timeout::Error] if @exceptions.blank?
+      @exceptions = [Timeout::Error] if @exceptions.empty?
```
Comment (yarmand) [inline lib/circuitbox/circuit_breaker.rb]: If there is not one already, I would add a test when `@exceptions = nil` to be sure the behavior is covered the same way by `empty?`

Comment (matthewshafer) [reply lib/circuitbox/circuit_breaker.rb]: If someone passes ```{ exceptions: nil }``` empty is going to fail with ```NoMethodError (undefined method `empty?' for nil:NilClass)```. ```blank?``` would have returned true. I can't come up with a reason to use circuitbox and pass in no exceptions to monitor so I'm file with leaving this how it is. Worst case we could do ```Array(@exceptions)``` but that just feels like allowing the API to be used incorrectly.

Comment (yarmand) [reply lib/circuitbox/circuit_breaker.rb]: I agree this would not be a "tell don't ask" situation here as the default on `Timeout::Error` is not obvious.
If we want to discourage bad API usage, I think we should then be explicit in the error.
```ruby
raise 'exceptions need to be an array' unless @exceptions.is_a?(Array)
```

Comment (matthewshafer) [reply lib/circuitbox/circuit_breaker.rb]: I think that makes sense. I'm wondering if we can do a little more cleanup (another PR) where we can move ```Timeout::Error``` to the options.fetch call otherwise if ```:exceptions``` are specified it needs to be given something that's not empty.
